.github/workflows/ci.yml

```diff
@@ -17,6 +17,7 @@ jobs:
           - windows-latest
           - ubuntu-latest
           - macos-latest
+          - macos-14
```

Comment:
What's the difference between this and the latest version above? Are both needed?

Reply:
well, according to https://github.blog/changelog/2024-01-30-github-actions-macos-14-sonoma-is-now-available/
macos-14 is running on m1 machines and macos-latest is now targeting macos12 and is running on x64 architecture

Reply:
`macos-latest` will become `macos-14` at least from my interpretation of the blog post

I believe we should pin to `macos-12` to ensure we retain x86_64 builds, when `macos-latest` switches

> The macOS 12 runner image will remain latest until migration of the latest YAML workflow label to macOS 14 in Q2 FY24 (April – June 2024). While macOS 13 is now generally available under the macos-13 label, this image will not be migrated to latest. Following this announcement, macOS 11 runner image will begin deprecation immediately with retirement expected to complete by June 2024.

The runner images show readme's for ARM/AMD versions of macos-14 but there doesn't seem to be a way to specify an arch (that I've seen), so if intel mac's get phased out on gh actions, we may have to rely on rosetta

We've done that against the other two repos currently using these runners

- https://github.com/pact-foundation/pact-php/pull/457
- https://github.com/pact-foundation/pact-js-core/pull/478